src/main.cpp

```diff
@@ -57,12 +59,18 @@ esp_err_t getIsConfigured(bool& is_configured)
     return ESP_OK;
 }
 
+
 extern "C" void app_main()
 {
     esp_err_t err;
+    wifi::WiFiEventHandler* event_handler;
+    event_handler = new wifi::WiFiEventHandler();
```

Comment:
Also this shouldn't be on the heap, you can pass the pointer using `wifi_mode->setWiFiEventHandler(&event_handler);` and removing the `delete` in `WiFiMode::~WiFiMode`. Also what about using a `std::unique_ptr` instead of manual memory allocation? `std::unique_ptr` is the same size of a pointer, doesn't add any overhead and we don't need to care about deallocating the memory, it gets destroyed automatically.